File: cryptography-cipher/src/main/java/dev/retrotv/crypto/cipher/block/BlockCipher.java
package dev.retrotv.crypto.cipher.block;

import dev.retrotv.crypto.cipher.enums.ECipher;
import lombok.Getter;

/**
 * 블록 암호화 알고리즘 구현을 위한 추상 클래스 입니다.
 *
 * @author yjj8353
 * @since 1.0.0
 */
@Getter
public abstract class BlockCipher {
    protected org.bouncycastle.crypto.BlockCipher engine;
    protected ECipher algorithm;
}

File: cryptography-cipher/src/main/kotlin/dev/retrotv/crypto/cipher/block/BlockCipher.kt
package dev.retrotv.crypto.cipher.block

import dev.retrotv.crypto.cipher.enums.ECipher
import org.bouncycastle.crypto.BlockCipher

/**
 * 블록 암호화 알고리즘 구현을 위한 추상 클래스 입니다.
 */
abstract class BlockCipher {
    lateinit var engine: BlockCipher
    lateinit var algorithm: ECipher
}